packages/format-library/src/non-breaking-space/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { insert } from '@wordpress/rich-text';
+import { RichTextShortcut } from '@wordpress/block-editor';
+import { isAppleOS } from '@wordpress/keycodes';
+
+const name = 'core/non-breaking-space';
+const title = __( 'Non breaking space' );
+
+export const nonBreakingSpace = {
+	name,
+	title,
+	tagName: 'span',
```

Comment:
Are we required to add the tagName? Maybe we should allow formats without this in the future.

Reply:
`tagNam`e is required by the following rule:

https://github.com/WordPress/gutenberg/blob/45701ad48fc1ac77baf8e905f6f84a87b940224b/packages/rich-text/src/register-format-type.js#L57-L59

Also, if `className` is not specified, it will be duplicated with the underline definition, so it is explicitly specified.

```
register-format-type.js:86 Format "core/underline" is already registered to handle bare tag name "span".
```

Reply:
I think it's fine to have it like this for now.

Reply:
Let's leave a comment to explain?